MapboxNavigation/Error.swift

```diff
@@ -5,10 +5,9 @@ extension NSError {
     /**
      Creates a custom `Error` object.
      */
-    public convenience init(localizedFailureReason: String, code: MapboxNavigationError = .defaultError) {
+    convenience init(code: MBErrorCode, localizedFailureReason: String, spokenInstructionCode: SpokenInstructionErrorCode? = nil) {
         self.init(domain: MBErrorDomain, code: code.rawValue, userInfo: [
-            NSLocalizedFailureReasonErrorKey: localizedFailureReason,
-            MBSpokenInstructionErrorCode: code
+            NSLocalizedFailureReasonErrorKey: localizedFailureReason
```

Comment:
`spokenInstructionCode` appears to be unused. Add an entry here that sets `MBSpokenInstructionErrorCodeKey` to `spokenInstructionCode`, but only if it’s provided.